### PR TITLE
Make PEtab an optional dependency (Closes #134)

### LINKED ISSUE #134: Disentangle petab from pypesto
pyPESTO should have as minimal requirements as possible: I can think of quite some cases, in which we might want to use pyPESTO without PEtab (which needs libsbml, which can in turn be problematic), so these things should not be necessarily linked.  In order to make pyPESTO usable also for non-AMICI and non-PEtab-users, we need to unlink those two things.

### PR DESCRIPTION
**Pull request by dweindl**

### CODE REVIEW

File: pypesto/objective/__init__.py [hunk 1]
```diff
@@ -7,10 +7,15 @@
...
 from .amici_objective import AmiciObjective
 from .aggregated import AggregatedObjective
-from .petab_import import PetabImporter
 from .options import ObjectiveOptions
 from .util import res_to_chi2, sres_to_schi2
 
+# PEtab is an optional dependency
+try:
+    from .petab_import import PetabImporter
+except ModuleNotFoundError:
```
Comment (dweindl) [inline pypesto/objective/__init__.py]: Codacy Issue found: [undefined name 'ModuleNotFoundError'](https://app.codacy.com/app/dweindl/pyPESTO/pullRequest?prid=4686900)

**Comment by dweindl** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/dweindl/pyPESTO/pullRequest?prid=4686900&bid=15671641)

**Comment by dweindl** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/dweindl/pyPESTO/pullRequest?prid=4686900&bid=15671641)

File: pypesto/objective/__init__.py (hunk 1 above)
Comment (dweindl) [inline pypesto/objective/__init__.py]: Codacy Issue found: [undefined name 'ModuleNotFoundError'](https://app.codacy.com/app/dweindl/pyPESTO/pullRequest?prid=4686900)

File: setup.py [hunk 2]
```diff
@@ -27,9 +27,9 @@ def read(fname):
                         'scipy>=1.1.0',
                         'pandas>=0.23.4',
                         'matplotlib>=2.2.3',
-                        'cloudpickle>=0.7.0',
```
Comment (yannikschaelte) [inline setup.py]: why remove?

Comment (dweindl) [reply setup.py]: Not removed, only the bracket moved.

Comment (yannikschaelte) [reply setup.py]: oh right. silly me.

File: pypesto/objective/__init__.py (hunk 1 above)
Comment (yannikschaelte) [reply pypesto/objective/__init__.py]: duh. ModuleNotFoundError was introduced in python 3. so, seems like codacy messes up? Either change to `ImportError`, or try adding a .codacy.yml file having
```yml
engines:
  pylint:
    enabled: true
    python_version: 3
```

Comment (dweindl) [reply pypesto/objective/__init__.py]: Thanks!

**Comment by dweindl** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/dweindl/pyPESTO/pullRequest?prid=4686900&bid=15671641)

**Comment by dweindl** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/dweindl/pyPESTO/pullRequest?prid=4686900&bid=15671641)

File: pypesto/objective/__init__.py (hunk 1 above)
Comment (dweindl) [inline pypesto/objective/__init__.py]: Codacy Issue found: [undefined name 'ModuleNotFoundError'](https://app.codacy.com/app/dweindl/pyPESTO/pullRequest?prid=4686900)

**Comment by dweindl** [comment]
Travis failures are related to upstream changes (benchmark collection), so merging anyways.

**Comment by dweindl** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/dweindl/pyPESTO/pullRequest?prid=4686900&bid=15671641)

File: pypesto/objective/__init__.py (hunk 1 above)
Comment (dweindl) [inline pypesto/objective/__init__.py]: Codacy Issue found: [undefined name 'ModuleNotFoundError'](https://app.codacy.com/app/dweindl/pyPESTO/pullRequest?prid=4686900)

**Comment by dweindl** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/dweindl/pyPESTO/pullRequest?prid=4686900&bid=15671641)

File: pypesto/objective/__init__.py (hunk 1 above)
Comment (dweindl) [inline pypesto/objective/__init__.py]: Codacy Issue found: [undefined name 'ModuleNotFoundError'](https://app.codacy.com/app/dweindl/pyPESTO/pullRequest?prid=4686900)

**Pull request merged**